annotation-utils/src/main/java/edu/isi/vista/annotationutils/SerializedCasConverter.java

```diff
@@ -0,0 +1,198 @@
+package edu.isi.vista.annotationutils;
+
+import static org.apache.uima.cas.impl.Serialization.deserializeCASComplete;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ImmutableMap;
+import edu.isi.nlp.parameters.Parameters;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.uima.UIMAException;
+import org.apache.uima.cas.CAS;
+import org.apache.uima.cas.CASException;
+import org.apache.uima.cas.impl.CASCompleteSerializer;
+import org.apache.uima.cas.impl.CASImpl;
+import org.apache.uima.fit.util.JCasUtil;
+import org.apache.uima.jcas.JCas;
+import org.apache.uima.resource.ResourceInitializationException;
+import org.apache.uima.resource.metadata.TypeSystemDescription;
+import org.apache.uima.util.CasCreationUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import webanno.custom.CTEventSpan;
+import webanno.custom.CTEventSpanType;
+
+public class SerializedCasConverter {
+
+  private static final String USAGE =
+      "SerializedCasConverter param_file\n"
+          + "\tparam file consists of :-separated key-value pairs\n"
+          + "\tThe required parameters are:\n"
+          + "\teventType: event type\n"
+          + "\tinputDirectory: the path to a directory with serialized CAS files (.ser)\n"
+          + "\toutputDirectory: the path to a directory where output files to be written to";
+
+  private static final Logger log = LoggerFactory.getLogger(SerializedCasConverter.class);
+
+  private static final String PARAM_EVENT_TYPE = "eventType";
+
+  private static final String PARAM_INPUT_DIRECTORY = "inputDirectory";
+
+  private static final String PARAM_OUTPUT_DIRECTORY = "outputDirectory";
+
+  public static void main(String[] args) throws IOException {
+
+    // get parameters from parameter file
+    Parameters parameters = null;
+
+    if (args.length == 1) {
+      parameters = Parameters.loadSerifStyle(new File(args[0]));
+    } else {
+      System.err.println(USAGE);
+      System.exit(1);
+    }
+
+    String eventType = parameters.getString(PARAM_EVENT_TYPE);
+    File inputDir = new File(parameters.getString(PARAM_INPUT_DIRECTORY));
```

Comment:
Prefer `parameters.getExistingDirectory(...)`

Reply:
And for output, `getCreatableDirectory()`. That will also let you dispense with all the checks below.

Reply:
Changed.